src/components/ModalSources.tsx

```diff
@@ -128,38 +131,38 @@ class AddSource extends React.Component {
       }
       case 'geojson_json': return {
         type: 'geojson',
-        cluster: source.cluster || false,
+        cluster: (source as GeoJSONSourceSpecification).cluster || false,
         data: {}
       }
       case 'tilejson_vector': return {
         type: 'vector',
-        url: source.url || `${protocol}//localhost:3000/tilejson.json`
+        url: (source as VectorSourceSpecification).url || `${protocol}//localhost:3000/tilejson.json`
       }
       case 'tilexyz_vector': return {
         type: 'vector',
-        tiles: source.tiles || [`${protocol}//localhost:3000/{x}/{y}/{z}.pbf`],
-        minZoom: source.minzoom || 0,
-        maxZoom: source.maxzoom || 14
+        tiles: (source as VectorSourceSpecification).tiles || [`${protocol}//localhost:3000/{x}/{y}/{z}.pbf`],
+        minzoom: (source as VectorSourceSpecification).minzoom || 0,
```

Comment:
I need to check somehow if this is a bug fix or a wrong change...

Reply:
Seems like this fixes a bug when the default zoom - it is 0 to 22 before the bug fix and 0 to 14 afterwards, so this is a correct change, the value of this is just amazing so early on.